Kudu.Core/Jobs/TriggeredJobsManager.cs

```diff
@@ -31,7 +32,18 @@ public TriggeredJobsManager(ITraceFactory traceFactory, IEnvironment environment
 
         public override IEnumerable<TriggeredJob> ListJobs()
         {
-            return ListJobsInternal();
+            var cache = TriggeredJobCache;
+            if (cache == null)
+            {
+                cache = ListJobsInternal();
+                TriggeredJobCache = cache;
+            }
+            return cache;
+        }
```

Comment:
I think what you could do is to move this logic to base class ListJobsInternal().   Then the cache object (static member) would also be in the base class.  The challenge is it might have to be declared as object and, when used, casted accordingly.  Make sense?